lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -81,8 +83,54 @@ def add_node_data(node)
     node.add_server_facts(@server_facts)
   end
 
+  # Rewrite a given file resource with the metadata from a fileserver based file
+  def replace_metadata(resource, metadata)
+    if metadata.ftype == "file"
+      resource[:checksum_value] = sumdata(metadata.checksum)
+    end
+  end
+
+  # Inline file metadata for static catalogs
+  # Initially restricted to files sourced from codedir via puppet:/// uri.
+  def inline_metadata(catalog, agent_checksum_type)
+    # Determine which checksum to use; if agent_checksum_type is not nil,
+    # use the first entry in it that is also in known_checksum_types.
+    # If no match is found, give an error.
+    if agent_checksum_type
+      agent_checksum_types = agent_checksum_type.split('.').map {|type| type.to_sym}
+      checksum_type = agent_checksum_types.drop_while do |type|
+        not known_checksum_types.include? type
+      end.first
+    end
+
+    raise Puppet::Error, "Unable to find a common checksum type between agent '#{agent_checksum_types}' and master '#{known_checksum_types}'." unless checksum_type
```

Comment:
If someone curls the endpoint and sets `static_catalog` but doesn't set `checksum_type` then we will enter this code, but fail on the "Unable to find a common checksum type..." Thinking it would better to fail the request sooner if only `static_catalog` is set, but `checksum_type` is not.